core-plugins/src/main/java/co/cask/hydrator/plugin/batch/file/AbstractFileCopySink.java

```diff
@@ -147,8 +147,16 @@ public FileCopyOutputFormatProvider(AbstractFileCopySink.AbstractFileCopySinkCon
       FileCopyOutputFormat.setBasePath(conf, config.basePath);
       FileCopyOutputFormat.setEnableOverwrite(conf, config.enableOverwrite.toString());
       FileCopyOutputFormat.setPreserveFileOwner(conf, config.preserveFileOwner.toString());
-      FileCopyOutputFormat.setBufferSize(conf, String.valueOf(config.bufferSize));
-      FileCopyOutputFormat.setFilesystemCaching(conf, config.filesystemCaching, config.getScheme());
+
+      if (config.bufferSize != null) {
+        // bufferSize is in megabytes
+        FileCopyOutputFormat.setBufferSize(conf, String.valueOf(config.bufferSize << 20));
+      } else {
+        FileCopyOutputFormat.setBufferSize(conf, String.valueOf(FileCopyRecordWriter.DEFAULT_BUFFER_SIZE));
+      }
+
+      // always disable caching
+      conf.put(String.format("fs.%s.impl.disable.cache", config.getScheme()), String.valueOf(true));
```

Comment:
We only want our specific OutputFormat to disable caching, not everything else in Hadoop. So we shouldn't set this here, but only in the OutputFormat when we get a new FileSystem object.